pandas/io/pytables.py

```diff
@@ -1038,6 +1038,37 @@ def groups(self):
                  g._v_name != u('table')))
         ]
 
+    def walk(self):
+        """ Walk the pytables group hierarchy yielding the group name and pandas object names
+        for each group.  Any non-pandas PyTables objects that are not a group will be ignored.
+
+        Returns
+        -------
+        A generator yielding tuples (`path`, `groups`, `frames`) where:
+
+        - `path` is the full path to a group,
+        - `groups` is a list of group names contained in `path`
+        - `leaves` is a list of pandas object names contained in `path`
+
+        """
+        _tables()
+        self._check_if_open()
+        for g in self._handle.walk_groups():
+            if getattr(g._v_attrs, 'pandas_type', None) is not None:
```

Comment:
No, HDF5 groups that have 'pandas_type' attribute will be group wrappers around dataframe/series objects.  Every Pandas object is wrapped in an HDF5 group.